<a href="https://colab.research.google.com/github/ksuaray/LAEP_S24/blob/Heart-Disease/New_Heart_Disease_Assignment_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Heart Disease**

#**Importing Necessary Python Modules**

Python incorporates a variety of open source add-ins called **modules** that enable us to be able to

In [ ]:
import pandas as pd
import numpy as np
import plotly.express as px
from IPython.display import Image


#**Context**

Coronary heart disease (CHD) involves the reduction of blood flow to the heart muscle due to the build-up of plaque in the arteries of the heart. It is the most common form of cardiovascular disease. Currently, invasive coronary angiography represents the gold standard for establishing the presence, location, and severity of CAD, however, this diagnostic method is costly and associated with morbidity and mortality in CAD patients. Therefore, it would be beneficial to develop a non-invasive alternative to replace the current gold standard.

Other less invasive diagnostics methods have been proposed in the scientific literature including exercise electrocardiogram, thallium scintigraphy, and fluoroscopy of coronary calcification. However, the diagnostic accuracy of these tests only ranges between 35%-75%. Therefore, it would be beneficial to develop a computer-aided diagnostic tool that could utilize the combined results of these non-invasive tests in conjunction with other patient attributes to boost the diagnostic power of these non-invasive methods with the aim of ultimately replacing the current invasive gold standard.

A number of 303 consecutive patients referred for coronary angiography at the Cleveland Clinic between May 1981 and September 1984 participated in the experiment. No patient had a history or electrocardiographic evidence of prior myocardial infarction or known valvular or cardiomyopathic diseases.


#**About the Dataset**

The dataset comprises 303 observations, 13 features, and 1 target attribute. The 13 features include the results of the aforementioned non-invasive diagnostic tests along with other relevant patient information. The target variable includes the result of the invasive coronary angiogram which represents the presence or absence of coronary artery disease in the patient. The 14 variables (13 features and 1 target attribute) are described below.

| **Item**    | **Description**                                                                                   |
|-------------|---------------------------------------------------------------------------------------------------|
| AGE         | Displays the age of the individual.                                                               |
| SEX         | Displays the gender of the individual using the following format: 1 = male, 0 = female.          |
| CP          | Displays the type of chest pain experienced by the individual:                                    |
|             | - 0 = typical angina                                                                             |
|             | - 1 = atypical angina                                                                            |
|             | - 2 = non-anginal pain                                                                           |
|             | - 3 = asymptotic                                                                                 |
| TRESTBPS    | Displays the resting blood pressure value of an individual in mmHg (unit).                        |
| CHOL        | Displays the serum cholesterol in mg/dl (unit).                                                  |
| FBS         | Compares the fasting blood sugar value of an individual with 120mg/dl:                             |
|             | - 1: fasting blood sugar >120mg/dl                                                               |
|             | - 0: fasting blood sugar ≤ 120mg/dl                                                              |
| RESTECG     | Displays resting electrocardiographic results:                                                   |
|             | - 0 = normal                                                                                     |
|             | - 1 = having ST-T wave abnormality                                                               |
|             | - 2 = left ventricular hypertrophy                                                               |
| THALACH     | Displays the max heart rate achieved by an individual.                                           |
| EXANG       | Exercise-induced angina:                                                                         |
|             | - 1 = yes                                                                                        |
|             | - 0 = no                                                                                         |
| OLDPEAK     | ST depression induced by exercise relative to rest. Displays the value (integer or float).       |
| SLOPE       | Peak exercise ST segment:                                                                        |
|             | - 1 = upsloping                                                                                  |
|             | - 2 = flat                                                                                       |
|             | - 3 = downsloping                                                                                |
| CA          | Number of major vessels (0-3) colored by fluoroscopy. Displays the value (integer or float).      |
| THAL        | Displays the thalassemia:                                                                       |
|             | - 1 = normal                                                                                     |
|             | - 2 = fixed defect                                                                              |
|             | - 3 = reversible defect                                                                         |
| TARGET      | Displays whether the individual is suffering from heart disease or not:                           |
|             | - 0 = absence                                                                                   |
|             | - 1 = present                                                                                   |





**Data**

In [ ]:
url="https://archive.ics.uci.edu/static/public/45/data.csv"
df=pd.read_csv(url)
df

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  \
0     63    1   1       145   233    1        2      150      0      2.3   
1     67    1   4       160   286    0        2      108      1      1.5   
2     67    1   4       120   229    0        2      129      1      2.6   
3     37    1   3       130   250    0        0      187      0      3.5   
4     41    0   2       130   204    0        2      172      0      1.4   
..   ...  ...  ..       ...   ...  ...      ...      ...    ...      ...   
298   45    1   1       110   264    0        0      132      0      1.2   
299   68    1   4       144   193    1        0      141      0      3.4   
300   57    1   4       130   131    0        0      115      1      1.2   
301   57    0   2       130   236    0        2      174      0      0.0   
302   38    1   3       138   175    0        0      173      0      0.0   

     slope   ca  thal  num  
0        3  0.0   6.0    0  
1        2  3.0   3.0    2  
2        2  2.0   7.0    1  
3        3  0.0   3.0    0  
4        1  0.0   3.0    0  
..     ...  ...   ...  ...  
298      2  0.0   7.0    1  
299      2  2.0   7.0    2  
300      2  1.0   7.0    3  
301      2  1.0   3.0    1  
302      1  NaN   3.0    0  

[303 rows x 14 columns]

Next, we can display the data by typing the name of the DataFrame. To ensure we can see all columns, we'll use the *pd.set_option* method.

In [ ]:
# Set display options to show all columns
pd.set_option('display.max_columns', None)
df

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  \
0     63    1   1       145   233    1        2      150      0      2.3   
1     67    1   4       160   286    0        2      108      1      1.5   
2     67    1   4       120   229    0        2      129      1      2.6   
3     37    1   3       130   250    0        0      187      0      3.5   
4     41    0   2       130   204    0        2      172      0      1.4   
..   ...  ...  ..       ...   ...  ...      ...      ...    ...      ...   
298   45    1   1       110   264    0        0      132      0      1.2   
299   68    1   4       144   193    1        0      141      0      3.4   
300   57    1   4       130   131    0        0      115      1      1.2   
301   57    0   2       130   236    0        2      174      0      0.0   
302   38    1   3       138   175    0        0      173      0      0.0   

     slope   ca  thal  num  
0        3  0.0   6.0    0  
1        2  3.0   3.0    2  
2        2  2.0   7.0    1  
3        3  0.0   3.0    0  
4        1  0.0   3.0    0  
..     ...  ...   ...  ...  
298      2  0.0   7.0    1  
299      2  2.0   7.0    2  
300      2  1.0   7.0    3  
301      2  1.0   3.0    1  
302      1  NaN   3.0    0  

[303 rows x 14 columns]

#**ASSIGNMENT 1 - Descriptive Statistics: Graphical and Numerical Summary**

**INSTRUCTIONS**

Use Python to analyze the data set and complete each of the following. As appropriate, copy the SPSS output and paste it in the correct part below. For problems that require a written response, type the answer below.

##**QUESTION 1**

Determine whether the three variables below are qualitative or quantitative. If they are quantitative, specify whether they are continuous or discrete.

| **Variable**              | **Type**          | **Description**                                      |
|---------------------------|-------------------|------------------------------------------------------|
| **Age**                   | Quantitative      | Continuous age of the individual.                    |
| **Sex**                   | Qualitative       | Gender of the individual (1 = male, 0 = female).     |
| **Chest Pain Type (CP)**  | Qualitative       | Type of chest pain experienced by the individual:    |
|                           |                   | - 0 = typical angina                                 |
|                           |                   | - 1 = atypical angina                                |
|                           |                   | - 2 = non-anginal pain                               |
|                           |                   | - 3 = asymptotic                                     |
| **Resting BP (TRESTBPS)** | Quantitative      | Resting blood pressure value in mmHg (unit).         |


##**QUESTION 2**

Construct a frequency table, relative frequency table, and relative frequency bar chart to describe the distribution of chest pain type. State any fact that jumps out to you.

In [ ]:
#Frequency table
freq_table = pd.value_counts(df['cp'])
freq_table

4    144
3     86
2     50
1     23
Name: cp, dtype: int64

In [ ]:
#Relative frequency table
freq_table/len(df)

4    0.475248
3    0.283828
2    0.165017
1    0.075908
Name: cp, dtype: float64

In [ ]:
frequency_table = df['cp'].value_counts()

# Convert index (chest pain types) to strings if they're not already
frequency_table.index = frequency_table.index.astype(str)

# Create the bar chart
fig = px.bar(x=frequency_table.index, y=frequency_table.values,
             title='Frequency Distribution Bar Chart',
             labels={'x':'Chest Pain Type', 'y':'Frequency'})
fig.show()

47% of patients have atypical angina; 16.5% of patients have atypical angina; 28.7% of patients have non anginal pain and 7.6% have asymptotic pain.

## **QUESTIONS 3-6**

For questions 3-6: Find your variable based on your last name and use that variable when answering questions #3 to #6.  

| **Last Name Variable** | **Description**                                      |
|------------------------|------------------------------------------------------|
| A-F                    | Resting blood pressure                    |
| G-M                    | Serum cholesterol                              |
| N-S                    | Maximum heart rate achieved                |
| T-Z                    | ST depression induced by exercise relative to rest  |



###**QUESTION 3**

Construct a histogram for your variable. Use Number of Intervals = 15.

In [ ]:
fig = px.histogram(x=df['trestbps'],nbins = 15)
fig.show()

In [ ]:
fig = px.histogram(x=df['chol'],nbins = 15)
fig.show()

In [ ]:
fig = px.histogram(x=df['thalach'],nbins = 15)
fig.show()

In [ ]:
fig = px.histogram(x=df['oldpeak'],nbins = 15)
fig.show()

###**QUESTION 4**

Construct a boxplot for your variable.  

In [ ]:
px.box(x=df['trestbps'])

In [ ]:
px.box(x=df['chol'])

In [ ]:
px.box(x=df['thalach'])

In [ ]:
px.box(x=df['oldpeak'])

###**QUESTION 5**

Calculate the following summary statistics for your variable: minimum, maximum, mean, median, standard deviation, Q1, and Q3. Paste the output below.

In [ ]:
descriptive_stats = df[['cp','trestbps','chol','thalach','oldpeak']].describe()
descriptive_stats


cp    trestbps        chol     thalach     oldpeak
count  303.000000  303.000000  303.000000  303.000000  303.000000
mean     3.158416  131.689769  246.693069  149.607261    1.039604
std      0.960126   17.599748   51.776918   22.875003    1.161075
min      1.000000   94.000000  126.000000   71.000000    0.000000
25%      3.000000  120.000000  211.000000  133.500000    0.000000
50%      3.000000  130.000000  241.000000  153.000000    0.800000
75%      4.000000  140.000000  275.000000  166.000000    1.600000
max      4.000000  200.000000  564.000000  202.000000    6.200000

###**QUESTION 6**

Use information from questions #3, #4, and #5 to describe your variable in terms of shape, center, spread, and outliers. Interpret your findings.

The distribution of blood pressure is skewed right. The median is 130 mmHg and IQR is 20 mmHg. There are 7 outliers.

The distribution of serum cholesterol is skewed right. The median is 240 mg/dl and IQR is 64 mg/dl. There are 5 outliers.

The distribution of the maximum heart rate achieved is skewed left. The median is 153 mmHg and IQR is 33 mmHg. There is 1 outlier.

The distribution of ST depression induced by exercise relative to rest is skewed right. The median is 0.8 and IQR is 1.6. There are 4 outliers.

##**QUESTION 7**

Calculate and state the mean age, mean resting blood pressure, and mean cholesterol for those with heart disease, and for those without heart disease. Compare the results.

In [ ]:
sec_means = df[['age', 'trestbps','chol']].mean()
sec_stat = pd.DataFrame(sec_means, columns=['Mean'])
sec_stat

Mean
age        54.438944
trestbps  131.689769
chol      246.693069

Patients without heart disease tended to be older, with higher blood pressure and cholesterol levels.

##**QUESTION 8**

Generate a paragraph of at least 100 words to address one of the following questions:

a) Discuss how analyzing your chosen data set using statistical methods could help you become better prepared for future courses in your major?

In upper division STAT courses, I frequently have to use descriptive statistics and visualization to display datasets. Knowing how to generate these graphs and tables can make it easier to grasp the dataset that I am working with. It is much easier to determine the skew, distribution, and behavior of data once the descriptive statistics are generated.

### **QUESTION 8a**

Discuss how analyzing your chosen data set using statistical methods could help you become better prepared for future courses in your major?

### **QUESTION 8b**

Discuss how analyzing your chosen data set using statistical methods could be instrumental in becoming better prepared for your future career?